File: app/src/main/java/com/kelsos/mbrc/data/Artist.java
package com.kelsos.mbrc.data;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.JsonNode;

public class Artist {

  @JsonProperty("artist") private String artist;

  @JsonProperty("count") private int count;

  public Artist() {

  }

  public Artist(JsonNode node) {
    this.artist = node.path("artist").textValue();
    this.count = node.path("count").intValue();
  }

  @JsonProperty("artist") public String getArtist() {
    return artist;
  }

  @JsonProperty("artist") public void setArtist(String artist) {
    this.artist = artist;
  }

  @SuppressWarnings("unused")
  @JsonProperty("count") public int getCount() {
    return count;
  }

  @JsonProperty("count") public void setCount(int count) {
    this.count = count;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/content/library/artists/Artist.kt
package com.kelsos.mbrc.content.library.artists

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import com.kelsos.mbrc.RemoteDatabase
import com.kelsos.mbrc.interfaces.data.Data
import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table

@Table(name = "artist", database = RemoteDatabase::class)
data class Artist(
    @JsonProperty("artist")
    @Column
    var artist: String? = null,
    @JsonProperty("count")
    @Column
    var count: Int = 0,
    @JsonIgnore
    @Column
    @PrimaryKey(autoincrement = true)
    var id: Long = 0
) : Data
